R/plot_section.R

```diff
@@ -0,0 +1,207 @@
+#' @param level \code{character}. The name of the column you wish to be treated
+#'   as the stratigraphic levels associated with each occurrence, e.g. "bed"
+#'   (default) or "height".
+#' @param certainty \code{character}. The name of the column you wish to be
+#'   treated as the information on whether an identification is certain ("1") or
+#'   uncertain ("0"). By default (\code{certainty = NULL}), no column name is
+#'   provided, and all occurrences are assumed to be certain. In the plot, any
+#'   occurrence labelled as certain will be plotted with a black circle, while
+#'   any occurrence labelled as uncertain will be plotted with a white circle.
+#' @param by \code{character}. How should the output be sorted? Either: "FAD"
+#'   (first appearance; default), "LAD" (last appearance), or "name"
+#'   (alphabetically by taxon names).
+#' @param xlab \code{character}. The x-axis label (over taxa) passed directly to
+#'   \code{plot}.
+#' @param ylab \code{character}. The y-axis label (over strata) passed directly
+#'   to \code{plot}.
+#' @param plot_args A list of optional arguments that are passed directly to
+#'   [graphics::plot()].
+#' @param x_args A list of optional arguments that are passed directly to
+#'   [axis()] when generating the x-axis.
+#' @param y_args A list of optional arguments that are passed directly to
+#'   [axis()] when generating the x-axis.
+#'
+#' @return Invisibly returns a data.frame of the calculated taxonomic
+#'   stratigraphic ranges.
+#'
+#'   Function is usually used for its side effect, which is to create a plot
+#'   showing the stratigraphic ranges of taxa in a section, with levels at which
+#'   the taxon was sampled indicated with a point.
+#'
+#' @section Developer(s): Bethany Allen & Alexander Dunhill
+#' @section Reviewer(s): William Gearty & Lewis A. Jones
+#' @importFrom graphics axis par segments plot points box
+#'
+#' @examples
+#' #Load tetrapod dataset
+#' data(tetrapods)
+#' # Sample tetrapod occurrences
+#' tetrapod_names <- tetrapods$accepted_name[1:50]
+#' # Simulate bed numbers
+#' beds_sampled <- sample.int(n = 10, size = 50, replace = TRUE)
+#' # Simulate certainty values
+#' certainty_sampled <- sample(x = 0:1, size = 50, replace = TRUE)
+#' #Combine into data frame
+#' occdf <- data.frame(taxon = tetrapod_names, bed = beds_sampled,
+#' certainty = certainty_sampled)
+#' # Plot stratigraphic ranges
+#' par(mar = c(12, 5, 2, 2))
+#' plot_section(occdf, ylab = "Stratigraphic height (m)")
+#' plot_section(occdf, certainty = "certainty",
+#'              ylab = "Stratigraphic height (m)")
+#' plot_section(occdf, certainty = "certainty", by = "LAD",
+#'              ylab = "Bed number")
+#' plot_section(occdf, certainty = "certainty", by = "name",
+#'              ylab = "Bed number")
+#' plot_section(occdf, certainty = "certainty", ylab = "Bed number",
+#'              plot_args = list(main = "Section A"))
+#'
+#' @export
+plot_section <- function(occdf, name = "taxon", level = "bed",
```

Comment:
Elsewhere in the package, the default for `name` is generally "genus", do we want to stick with this?